apps/full-stack-tests/src/hierarchies/learning-snippets/Grouping.test.ts

```diff
@@ -0,0 +1,788 @@
+        });
+      });
+
+      describe("By properties", () => {
+        it("groups by property value", async function () {
+          const { imodel } = await buildIModel(this, async (builder) => {
+            insertRepositoryLink({ builder, repositoryLabel: "Example iModel link 1", format: "iModel" });
+            insertRepositoryLink({ builder, repositoryLabel: "Example iModel link 2", format: "iModel" });
+            insertRepositoryLink({ builder, repositoryLabel: "Example DGN link", format: "DGN" });
+            insertRepositoryLink({ builder, repositoryLabel: "Example link with no format" });
+          });
+          const imodelAccess = createIModelAccess(imodel);
+
+          // __PUBLISH_EXTRACT_START__ Presentation.Hierarchies.Grouping.PropertyValueGroupingExample
+          const hierarchyProvider = createHierarchyProvider({
+            imodelAccess,
+            hierarchyDefinition: {
+              defineHierarchyLevel: async ({ parentNode }) => {
+                if (!parentNode) {
+                  // The hierarchy definition requests `BisCore.RepositoryLink` nodes to be returned as root nodes and have
+                  // them grouped by the `Format` property.
+                  return [
+                    {
+                      fullClassName: "BisCore.RepositoryLink",
+                      query: {
+                        ecsql: `
+                          SELECT ${await createNodesQueryClauseFactory({ imodelAccess }).createSelectClause({
+                            ecClassId: { selector: "this.ECClassId" },
+                            ecInstanceId: { selector: "this.ECInstanceId" },
+                            nodeLabel: { selector: "this.UserLabel" },
+                            grouping: {
+                              byProperties: {
+                                propertiesClassName: "BisCore.RepositoryLink",
+                                propertyGroups: [
+                                  {
+                                    propertyClassAlias: "this",
+                                    propertyName: "Format",
+                                  },
+                                ],
+                                // create a grouping node for instances whose `Format` property value is not specified
+                                createGroupForUnspecifiedValues: true,
+                              },
+                            },
+                          })}
+                          FROM BisCore.RepositoryLink this
+                        `,
+                      },
+                    },
+                  ];
+                }
+                return [];
+              },
+            },
+          });
+
+          // The iModel has three elements of `BisCore.RepositoryLink` class - two with `iModel` and one with `DGN`
+          // `Format` property value. As requested by hierarchy definition, the provider returns them grouped
+          // the property value:
```

Comment:
```suggestion
        // The iModel has four elements of `BisCore.RepositoryLink` class:
        //
        // | Element's label             | `Format` property value |
        // | --------------------------- | ----------------------- |
        // | Example iModel link 1       | iModel                  |
        // | Example iModel link 2       | iModel                  |
        // | Example DGN link            | DGN                     |
        // | Example link with no format |                         |
        //
        // As requested by hierarchy definition, the provider returns them grouped by `Format` property value:
```

I guess Readme.md should also be updated.

Reply:
good catch, fixed